src/ja/common/proxy/command_handler.py

```diff
@@ -17,7 +22,45 @@ class CommandHandler(ABC):
     success property of the Response is True if both validations were passed
     and the Command was executed without error, and is false otherwise.
     """
-    def __init__(self, socket: str):
+    def __init__(self, socket_path: str):
         """!
-        @param socket: The Unix named socket to listen for Commands on.
+        @param socket_path: The Unix named socket to listen for Commands on.
         """
+        self._socket_path = socket_path
+        self._listen_thread = Thread(target=self._listen)
+        self._listen_thread.daemon = True  # Ensures that the thread is killed when main thread exits
+        self._listen_thread.start()
+
+    def _listen(self) -> None:
+        # Make sure the socket does not already exist
+        try:
+            os.unlink(self._socket_path)
+        except OSError:
+            if os.path.exists(self._socket_path):
+                raise
+        named_socket = socket.socket(family=socket.AF_UNIX, type=socket.SOCK_STREAM)
+        named_socket.bind(self._socket_path)
+        named_socket.listen(1)
+
+        while True:
+            connection, client_address = named_socket.accept()
+            try:
+                # First 8 bytes encode command length:
+                command_length = int.from_bytes(bytes=connection.recv(8), byteorder="big")
```

Comment:
Is recv guaranteed to give us 8 bytes? As far as I understand the documentation, you could also get 2 bytes and the next 6 bytes later ...

Reply:
Well, you also have no guarantee to ever receive a special character that signals the end of the input or that the socket will ever be closed.

Reply:
Well I think we will receive everything since it is our own remote that is sending us those messages (so we shouldn't get malicious input)

But the problem is rather, the system is free to send it in chunks of 2 bytes, and we won't handle that (I imagine you could write a small function to read N bytes, which does something like `len(buffer) < N: buffer += recv(N-len(buffer))`